File: examples/testing-java-with-jruby/test/greeting_test.rb
require "java"
require "test/unit"

java_import "Greeting"

class GreetingTest < Test::Unit::TestCase
  def setup
    @greeting = Greeting.new "Mateus Linhares", "mateus@idopter.com"
    @date = Time.now.strftime "%d/%m/%Y"
  end

  def test_say_hello
    assert_equal "Hello, Mateus Linhares. Your email: mateus@idopter.com. Date: #{@date}", @greeting.say_hello
  end

  def test_say_hello_with_assigns
    @greeting.name = "Piguera"
    @greeting.email = "pigor@idopter.com"
    assert_equal "Hello, Piguera. Your email: pigor@idopter.com. Date: #{@date}", @greeting.say_hello
  end
end

File: examples/testing-java-with-jruby/tests/GreetingTest.java
import static org.junit.Assert.assertEquals;

import java.text.SimpleDateFormat;
import java.util.Date;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class GreetingTest {
	Greeting greeting;
	String date;

	@Test
	public void testSayHello() {
		greeting = new Greeting("Mateus", "mateus@idopter.com");
		date = new SimpleDateFormat("dd/MM/yyyy").format(new Date());
		assertEquals("Hello, Mateus. Your email: mateus@idopter.com. Date: " + date, greeting.sayHello());
	}

}
